### PR TITLE
Add simulation train notebook, fix config

### PR DESCRIPTION
**Pull request by lucabergamini**
Add simulation train notebook (no distributed training, copied over from planning)

### CODE REVIEW

File: examples/simulation/config.yaml [hunk 1]
```diff
@@ -55,7 +55,7 @@ train_data_loader:
   batch_size: 12
```
Comment (yawei-ye) [inline examples/simulation/config.yaml:55]: Shall we curate 10hr training set for notebook training?

File: examples/simulation/train.ipynb [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "model = PlanningModel(\n",
+    "        model_arch=\"resnet50\",\n",
```
Comment (yawei-ye) [inline examples/simulation/train.ipynb:118]: read from config

File: examples/simulation/train.ipynb [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+    "        num_input_channels=rasterizer.num_channels(),\n",
+    "        num_targets=3 * cfg[\"model_params\"][\"future_num_frames\"],  # X, Y, Yaw * number of future states,\n",
+    "        weights_scaling= [1., 1., 1.],\n",
```
Comment (yawei-ye) [inline examples/simulation/train.ipynb:121]: Move to config and rename it to `target_weights_scaling`

File: examples/simulation/train.ipynb [hunk 4]
```diff
@@ -0,0 +1,289 @@
...
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "train_cfg = cfg[\"train_data_loader\"]\n",
+    "train_dataloader = DataLoader(train_dataset, shuffle=train_cfg[\"shuffle\"], batch_size=train_cfg[\"batch_size\"], \n",
+    "                             num_workers=train_cfg[\"num_workers\"])\n",
+    "device = torch.device(\"cuda:0\" if torch.cuda.is_available() else \"cpu\")\n",
+    "model = model.to(device)\n",
+    "optimizer = optim.Adam(model.parameters(), lr=1e-3)\n",
```
Comment (yawei-ye) [inline examples/simulation/train.ipynb:146]: lr dropping with increased epoch number? At least add a TODO

File: examples/simulation/train.ipynb [hunk 5]
```diff
@@ -0,0 +1,289 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "tr_it = iter(train_dataloader)\n",
+    "progress_bar = tqdm(range(cfg[\"train_params\"][\"max_num_steps\"]))\n",
```
Comment (yawei-ye) [inline examples/simulation/train.ipynb:166]: Any plan to switch to epoch based training?

**Review by yawei-ye — COMMENTED** [review]
nice work! some comments, otherwise LGTM

File: examples/simulation/config.yaml (hunk 1 above)
Comment (yawei-ye) [inline examples/simulation/config.yaml:55]: Shall we curate a 10hr training set for notebook example?

File: examples/simulation/train.ipynb (hunk 5 above)
Comment (lucabergamini) [reply examples/simulation/train.ipynb:166]: nope, this is just intended as an example for training

File: examples/simulation/config.yaml (hunk 1 above)
Comment (lucabergamini) [reply examples/simulation/config.yaml:55]: I don't think it's required right now..maybe we can refactor this later to include many changes

File: examples/simulation/train.ipynb (hunk 4 above)
Comment (lucabergamini) [reply examples/simulation/train.ipynb:146]: add a note in the cell before training

**Review by yawei-ye — APPROVED** [review]
thanks for addr comments, LGTM

**Review by yawei-ye — APPROVED** [review]

File: examples/simulation/train.ipynb [hunk 6]
```diff
@@ -0,0 +1,297 @@
...
+    "Then, we load our config file with relative paths and other configurations (rasteriser, training params...)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#@title Download L5 Sample Dataset and install L5Kit\n",
```
Comment (stefanopini) [inline examples/simulation/train.ipynb:64]: typo?

File: examples/simulation/train.ipynb [hunk 7]
```diff
@@ -0,0 +1,297 @@
...
+    "- by using data from multiple sources you're **including much more variability in your training data**;\n",
+    "- two agents may have taken different choices at the same intersection, leading to **multi-modal data**;\n",
+    "- the **quality of the annotated data is expected to be sensibility lower** compared to the SDV, as we're leveraging a perception system.\n",
+    "\n",
+    "Still, the final prize is even better than planning as this policy can potentially drive all the agents in the scene and it's not limited to the SDV only.\n",
+    "\n",
+    "![simulation-example](https://github.com/lyft/l5kit/blob/master/images/simulation/simulation_example.svg?raw=1)\n"
+   ]
+  },
+  {
```
Comment (stefanopini) [inline examples/simulation/train.ipynb:22]: You may add a title like "Imports"

File: examples/simulation/train.ipynb [hunk 8]
```diff
@@ -0,0 +1,297 @@
...
+   "source": [
+    "# rasterisation\n",
+    "rasterizer = build_rasterizer(cfg, dm)\n",
+    "\n",
+    "# ===== INIT DATASET\n",
+    "train_zarr = ChunkedDataset(dm.require(cfg[\"train_data_loader\"][\"key\"])).open()\n",
+    "train_dataset = AgentDataset(cfg, train_zarr, rasterizer)\n",
+    "\n",
+    "# plot some examples\n",
+    "for idx in range(0, len(train_dataset), len(train_dataset) // 10):\n",
```
Comment (stefanopini) [inline examples/simulation/train.ipynb:101]: How big is the train dataset? Is a tenth of it reasonably small?

**Review by stefanopini — APPROVED** [review]
Thanks Luca, LGTM! Just minors

**Pull request merged**